### PR TITLE
feat(virtctl): Allow inference from registry or snapshot volume imports

### LINKED ISSUE #12784: virtctl create vm does not infer from DataVolumes that import from a source other than a PVC
**What happened**:

Run the following command:

```
virtctl create vm --volume-import 'type:registry,url:docker://quay.io/containerdisks/centos-stream:9,size:10Gi'
```

In the created VM `spec.instancetype` and `spec.preference` are not set.

`virtctl create vm` only enables inference from DV's cloning a PVC or DS source, see [1].

[1] https://github.com/kubevirt/kubevirt/blob/1a50ec56c184c25fcb4e88777e3dd3e4c184daee/pkg/virtctl/create/vm/vm.go#L382-L388

**What you expected to happen**:

Run the following command:

```
virtctl create vm --volume-import 'type:registry,url:docker://quay.io/containerdisks/centos-stream:9,size:10Gi'
```

`spec.instancetype` and `spec.preference` are set to infer from the imported volume.

**How to reproduce it (as minimally and precisely as possible)**:
Steps to reproduce the behavior.

**Additional context**:

**Environment**:
- KubeVirt version (use `virtctl version`): N/A
- Kubernetes version (use `kubectl version`): N/A
- VM or VMI specifications: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from /etc/os-release): N/A
- Kernel (e.g. `uname -a`): N/A
- Install tools: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

This change enables the inference from registry or snapshot sources when creating a VM with virtctl create vm and the --volume-import flag.

Before this PR:

Inference is possible only from PVC sources.

After this PR:

Inference is possible from PVC, Registry and Snapshot sources.

Fixes #12784 

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
virtctl: Created VMs can infer an instancetype or preference from PVC, Registry and Snapshot sources now.
```

### CODE REVIEW

**Review by lyarwood — COMMENTED** [review]
/lgtm

File: pkg/virtctl/create/vm/vm.go [hunk 1]
```diff
@@ -376,21 +377,18 @@ func volumeValidToInferFrom(vm *v1.VirtualMachine, vol *v1.Volume) error {
...
-func dataVolumeValidToInferFrom(vm *v1.VirtualMachine, vol *v1.Volume) error {
+func dataVolumeValidToInferFrom(vm *v1.VirtualMachine, name string) error {
 	for _, dvt := range vm.Spec.DataVolumeTemplates {
-		if dvt.Name == vol.DataVolume.Name {
-			if dvt.Spec.Source != nil && dvt.Spec.Source.PVC != nil {
+		if dvt.Name == name {
+			if dvt.Spec.Source != nil && (dvt.Spec.Source.PVC != nil || dvt.Spec.Source.Registry != nil || dvt.Spec.Source.Snapshot != nil) {
 				return nil
 			}
-
```
Comment (alicefr) [inline pkg/virtctl/create/vm/vm.go:385]: nit: deleted empty space

File: pkg/virtctl/create/vm/vm.go [hunk 2]
```diff
@@ -376,21 +377,18 @@ func volumeValidToInferFrom(vm *v1.VirtualMachine, vol *v1.Volume) error {
...
 			if dvt.Spec.SourceRef != nil && dvt.Spec.SourceRef.Kind == "DataSource" {
 				return nil
 			}
-
-			return fmt.Errorf(InvalidInferenceVolumeError)
+			return fmt.Errorf(DVInvalidInferenceVolumeError)
 		}
 	}
-
```
Comment (alicefr) [inline pkg/virtctl/create/vm/vm.go:393]: Same

**Pull request merged**